src/data-workspace/data-workspace.test.js

```diff
@@ -0,0 +1,85 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+import { useWorkflowContext } from '../workflow-context/index.js'
+import { useSelectionParams } from '../workflow-context/use-selection-params.js'
+import { DataSetNavigation } from './data-set-navigation/index.js'
+import { DataWorkspace } from './data-workspace.js'
+import { Display } from './display/index.js'
+import { TitleBar } from './title-bar/index.js'
+
+jest.mock('../workflow-context/index.js', () => ({
+    useWorkflowContext: jest.fn(),
+}))
+
+jest.mock('../workflow-context/use-selection-params.js', () => ({
+    useSelectionParams: jest.fn(),
+}))
+
+const mockDataSets = [
+    {
+        id: 'data-set-1',
+        displayName: 'Data set 1',
+        periodType: 'Monthly',
+    },
+    {
+        id: 'data-set-2',
+        displayName: 'Data set 2',
+        periodType: 'Monthly',
+    },
+]
+
+const mockUseWorkflowContext = (workflow = {}) => {
+    useWorkflowContext.mockImplementation(() => ({
+        displayName: 'Workflow a',
+        id: 'i5m0JPw4DQi',
+        periodType: 'Monthly',
+        dataSets: [],
+        approvalState: 'APPROVED_HERE',
+        ...workflow,
+    }))
+}
```

Comment:
Generally I'm not a big fan of this (anymore). It makes test code much harder to parse, which reduces it's act-as-documentation capability. Some time ago Ismay shared a good article about abstraction in tests: [Why Good Developers Write Bad Unit Tests](https://mtlynch.io/good-developers-bad-tests/)

Reply:
Yeah I'm not a fan of mocking (context) hooks like this either but this is what we've been doing in existing tests so I thought I should follow the current style.

I think the two main alternatives to this approach are:
1. Wrap the `<DataWorkspace />` being tested in a provider, to which we pass in the mocked data (as is done for `Display` with `CustomDataProvider` below)
2. Split `DataWorkspace` into two and only test first component:
```js
const DataWorkspace = ({ periodId, orgUnitId, workflow, selectedDataSet, setSelectedDataSet }) => (
  <>
    <TitleBar name={workflow.name} dataSetsCount={/* ... */} />
    {/* ... */}
  </>
)

const DataWorkspaceWrapper = () => {
    const { pe: periodId, ou: organisationUnitId } = useSelectionParams()
    const workflow = useWorkflowContext()
    const [selectedDataSet, setSelectedDataSet] = useState(
        workflow.dataSets.length === 1 ? workflow.dataSets[0].id : null
    )

    return <DataWorkspace periodId={periodId} {/* ... */} />
}
```

Reply:
I think you could mock the implementation of the `useWorkflowContext` hook with default data and then, when different data is needed, use `mockImplementationOnce` and simply provide all data